ParProcCo/msmapper_aggregator.py

```diff
@@ -0,0 +1,268 @@
+            signal_name = nxdata.attrs["signal"]
+            self.signal_name = decode_to_string(signal_name)
+        elif "data" in nxdata.keys():
+            self.signal_name = "data"
+
+        if hasattr(self, "signal_name"):
+            if "axes" in nxdata.attrs:
+                self.axes_names = [decode_to_string(name) for name in nxdata.attrs["axes"]]
+            else:
+                self._generate_axes_names(nxdata)
+        else:
+            raise KeyError
+
+    def _generate_axes_names(self, nxdata: h5py.Group) -> None:
+        self.use_default_axes = True
+        signal_shape = nxdata[self.signal_name].shape
+        self.axes_names = [f"{letter}-axis" for letter in string.ascii_lowercase[:len(signal_shape)]]
+
+    def _get_all_axes(self) -> None:
+        self.signal_shapes = []
+        self.all_axes = []
+        for data_file in self.output_data_files:
+            with h5py.File(data_file, "r") as f:
+                signal_shape = f[self.nxdata_path_name][self.signal_name].shape
+                self.signal_shapes.append(signal_shape)
+                if self.renormalisation:
+                    weights_shape = f[self.nxdata_path_name]["weight"].shape
+                    assert signal_shape == weights_shape, "signal_shape must equal weights_shape"
+                if self.use_default_axes:
+                    axes = [list(range(length)) for length in signal_shape]
+                else:
+                    axes = [list(f[self.nxdata_path_name][axis_name]) for axis_name in self.axes_names]
+                self.all_axes.append(axes)
+        self.axes_spacing = [round((axis[1] - axis[0]), 6) for axis in axes]
+
+    def _accumulate_volumes(self) -> None:
+        for data_file, slices in zip(self.output_data_files, self.all_slices):
+            with h5py.File(data_file, "r") as f:
+                volumes_array = np.array(f[self.nxdata_path_name][self.signal_name])
+                if self.renormalisation:
+                    weights_array = np.array(f[self.nxdata_path_name]["weight"])
+
+            if self.renormalisation:
+                volumes_array = np.multiply(volumes_array, weights_array)
+                self.accumulator_weights[slices] += weights_array
+            self.accumulator_volume[slices] += volumes_array
+
+        if self.renormalisation:
+            self.total_volume = self.accumulator_volume / self.accumulator_weights
+            self.total_volume[np.isnan(self.total_volume)] = 0
+
+    def _write_aggregation_file(self, aggregation_output_dir: Path, output_data_files: List[Path]) -> Path:
+        aggregated_data_file = aggregation_output_dir / "aggregated_results.nxs"
+
+        with h5py.File(aggregated_data_file, "w") as f:
+            processed = f.create_group(self.nxentry_name)
+            processed.attrs["NX_class"] = "NXentry"
+            processed.attrs["default"] = self.nxdata_name
+
+            process_name = "/".join((self.nxentry_name, "process"))
```

Comment:
Rather than form absolute paths, you can use `create_group` and `create_dataset` on the groups. This will save all the `join` calls.